packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -204,6 +209,16 @@ function DashboardLayout(props: DashboardLayoutProps) {
 
   const branding = React.useContext(BrandingContext);
   const navigation = React.useContext(NavigationContext);
+  const { paletteMode, setPaletteMode, isDualTheme } = React.useContext(PaletteModeContext);
+
+  const [hasMounted, setHasMounted] = React.useState(false);
```

Comment:
Can you use `useSsr` from `@toolpad/utils`? It would avoid double renders when mounting client side.